src/processor/operator/aggregate/hash_aggregate_scan.cpp

```diff
@@ -37,5 +37,15 @@ bool HashAggregateScan::getNextTuplesInternal(ExecutionContext* /*context*/) {
     return true;
 }
 
+double HashAggregateScan::getProgress(ExecutionContext* /*context*/) const {
+    uint64_t totalNumFlatTuples = sharedState->getFactorizedTable()->getTotalNumFlatTuples();
```

Comment:
Let's use `getNumTuples`. `getTotalNumFlatTuples` has an overhead and is not meant to be called at runtime repeatedly. Double check if you have used this function in somewhere else.